crates/supervisor/service/src/actors/l1_watcher_rpc.rs

```diff
@@ -0,0 +1,222 @@
+//! [SupervisorActor] implementation for an L1 chain watcher that checks for L1 head updates over
+//! RPC.
+
+use crate::actors::traits::SupervisorActor;
+use alloy_eips::BlockId;
+use alloy_primitives::B256;
+use alloy_provider::{Provider, RootProvider};
+use async_trait::async_trait;
+use futures::StreamExt;
+use kona_genesis::RollupConfig;
+use kona_protocol::BlockInfo;
+use kona_rpc::{L1State, L1WatcherQueries};
+use std::sync::Arc;
+use thiserror::Error;
+use tokio::{
+    select,
+    sync::mpsc::{UnboundedSender, error::SendError},
+    task::JoinHandle,
+};
+use tokio_util::sync::CancellationToken;
+use tracing::{error, info, warn};
+
+/// An L1 chain watcher that checks for L1 head updates over RPC.
+#[derive(Debug)]
+pub struct L1WatcherRpc {
+    /// The [`RollupConfig`] to tell if ecotone is active.
+    /// This is used to determine if the L1 watcher should
+    /// check for unsafe block signer updates.
+    config: Arc<RollupConfig>,
+    /// The L1 provider.
+    l1_provider: RootProvider,
+    /// The latest L1 head sent to the derivation pipeline and watcher. Can be subscribed to, in
+    /// order to get the state from the external watcher.
+    latest_head: tokio::sync::watch::Sender<Option<BlockInfo>>,
```

Comment:
think it's fine just to know it's a watch channel, so can import that module in the prelude
```suggestion
    latest_head: watch::Sender<Option<BlockInfo>>,
```

Reply:
makes sense, changed it!